include/tvm/expr_operator.h

```diff
@@ -551,6 +557,12 @@ inline Expr MakeConstScalar(Type t, ValueType value) {
   if (t.is_int()) return ir::IntImm::make(t, static_cast<int64_t>(value));
   if (t.is_uint()) return ir::UIntImm::make(t, static_cast<uint64_t>(value));
   if (t.is_float()) return ir::FloatImm::make(t, static_cast<double>(value));
+  // For now, we store const scalar values of custom datatypes within doubles; later, during the
+  // datatypes lowering pass, we will lower the value to its true representation in the format
+  // specified by the datatype.
+  // TODO(gus) when do we need to start worrying about doubles not being precise enough?
+  if (tvm::runtime::GetCustomTypeRegistered(t.code()))
+    return ir::FloatImm::make(t, static_cast<double>(value));
```

Comment:
Perhaps we can simply ignore the check and directly return, assuming t exists, as long as t's code is bigger than custombegin